test/firewall.js

```diff
@@ -198,6 +198,24 @@ describe('Firewall', function() {
       firewall.setMetadata(metadata, assert.ifError);
     });
 
+    it('should respect network specification', function(done) {
+      var metadata = {network: 'custom-network'};
+
+      firewall.request = function(reqOpts) {
+        assert.strictEqual(reqOpts.method, 'PATCH');
```

Comment:
Sorry I missed this before. We can condense these assertions down to just one:

```js
assert.strictEqual(reqOpts.json.network, metadata.network);
```

Reply:
I still think that check should be done for all required fields. I.e. the current checks will guarantee that the payload we send to API endpoint is valid. 

I mean, for example, the use case when due to code error if metadata is not empty - required fields are set properly. E.g. that when I  `network` field is set - required field `name` field is set as well. E.g. imagine the code block like (oversimplified) that will lead to missing `name` field:

```
if metadata.network {
  metadata.network = metadata.network;
} else {
  metadata.name = this.name;
  metadata.network = this.metadata.network;
}
```

Reply:
I think we agree, the only thing is that I don't want to duplicate tests. In the previous test block, "it should make the correct API request", we check that all the arguments to `firewall.request()` are as expected. We don't need to confirm them again, since `src/firewall.js` only affects the `reqOpts.json.network` property when it sees `metadata.network` is set.

Reply:
I agree that maybe checks for request method and uri are excessive, but still in REST API specification is stated, that `name` is required parameter, so should be checked.

https://cloud.google.com/compute/docs/reference/rest/v1/firewalls/patch

Reply:
We don't need to check that `name` is defined in this test block ("should respect network specification"), however, since we know it is already set from the test above, "should make the correct API request".